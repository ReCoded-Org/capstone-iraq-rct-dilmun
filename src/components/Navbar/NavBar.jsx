import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { NavLink, useHistory } from 'react-router-dom'
import i18n from 'i18next'
import { useSelector, useDispatch } from 'react-redux'
import firebase from '../../firebase'
import Logo from '../../logo.svg'
import {
  HOME_ROUTE,
  PROFILE_ROUTE,
  CONTACT_ROUTE,
  ABOUT_ROUTE,
} from '../../router'
import Login from '../LogIn'
import userIcon from '../../assets/userIcon.png'
import { lOGOUT } from '../../redux/Authentication/ActionTypes'
import { LogOut } from '../../redux/Authentication/AuthenticationActions'

export default function Navbra() {
  const dispatch = useDispatch()
  const user = useSelector(state => state.authentication)

  const history = useHistory()
  const [profileDropDown, setprofileDropDown] = useState(false)
  const [navCollapse, setnavCollapse] = useState(false)

  const { t } = useTranslation()
  const handleLanguageChange = lang => {
    i18n.changeLanguage(lang)
  }

  const signOut = () => {
    firebase.auth().signOut()
    setprofileDropDown(false)
    dispatch(LogOut(lOGOUT))
    localStorage.removeItem('loggedInUser')
    history.push('/')
  }
  return (
    <>
      <nav className="bg-white z-50 sticky top-0  ">
        <div className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8">
          <div className="relative flex items-center justify-between h-16">
            <div className="absolute inset-y-0 left-0 flex items-center sm:hidden">
              <button
                type="button"
                className="inline-flex items-center justify-center p-2 rounded-md text-blue hover:text-darkBlue  focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                aria-controls="mobile-menu"
                aria-expanded="false"
                onClick={() => setnavCollapse(!navCollapse)}
              >
                <span className="sr-only">Open main menu</span>

                <svg
                  className="block h-6 w-6"
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke="currentColor"
                  aria-hidden="true"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    d="M4 6h16M4 12h16M4 18h16"
                  />
                </svg>

                <svg
                  className="hidden h-6 w-6"
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke="currentColor"
                  aria-hidden="true"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    d="M6 18L18 6M6 6l12 12"
                  />
                </svg>
              </button>
            </div>
            <div className="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start">
              <div className="flex-shrink-0 flex items-center">
                <img
                  className="block lg:hidden h-8 w-auto"
                  src={Logo}
                  alt="logo"
                />
                <img
                  className="hidden lg:block h-8 w-auto"
                  src={Logo}
                  alt="logo"
                />
              </div>
              <div className="hidden sm:block sm:ml-6">
                <div className="flex space-x-4 text-blue">
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    exact
                    to={HOME_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('navbar.home')}
                  </NavLink>
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    to={CONTACT_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('navbar.contact')}
                  </NavLink>
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    to={ABOUT_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('navbar.about')}
                  </NavLink>
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    to={PROFILE_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('footer.profile')}
                  </NavLink>
                </div>
              </div>
            </div>
            <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0">
              <button
                onClick={() => handleLanguageChange(t('language'))}
                type="button"
                className=" hover:bg-darkBlue hover:text-white items-center px-3   text-blue  rounded inline-flex focus:outline-none items-center border border-solid border-blue justify-center"
              >
                <i className="fas fa-globe pr-1" />
                <div className="text-center">{t('langText')}</div>
              </button>

              <div className="ml-3 relative">
                <div>
                  {user.isLoggedIn ? (
                    <button
                      onClick={() => setprofileDropDown(!profileDropDown)}
                      type="button"
                      className="bg-gray-800 flex text-sm rounded-full focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                      id="user-menu"
                      aria-expanded="false"
                      aria-haspopup="true"
                    >
                      <span className="sr-only">Open user menu</span>

                      <img
                        className="h-8 w-8 rounded-full"
                        src={user ? user.user.photo : userIcon}
                        alt=""
                      />
                    </button>
                  ) : (
                    <Login />
                  )}
                </div>

                <div
                  className={`${
                    profileDropDown ? ' show' : ' hidden'
                  } origin-top-right absolute right-0 border border-t-0 boder-grey mt-2 w-48 rounded-md items-center shadow-lg py-1 bg-white ring-1 ring-black ring-opacity-5 focus:outline-none`}
                  role="menu"
                  aria-orientation="vertical"
                  aria-labelledby="user-menu"
                >
                  <NavLink
                    to={PROFILE_ROUTE}
                    className="block px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white"
                    role="menuitem"
                  >
                    {user.isLoggedIn ? user.user.name : ''}
                  </NavLink>
                  <NavLink
                    to="/"
                    className="block px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white "
                    role="menuitem"
                  >
                    Settings
                  </NavLink>
                  <button
                    type="button"
                    onClick={signOut}
                    className="m-auto text-left block focus:outline-none px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white w-full"
                    role="menuitem"
                  >
                    Sign out
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>

        <div
          className={`${navCollapse ? 'show' : 'hidden'} sm:hidden`}
          id="mobile-menu"
        >
          <div className="px-2 pt-2 pb-3 space-y-1 text-blue">
            <NavLink
              to={HOME_ROUTE}
              className="text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
            >
              Home
            </NavLink>
            <NavLink
              to={CONTACT_ROUTE}
              className="text-blue  hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
            >
              Contact
            </NavLink>
            <NavLink
              to={ABOUT_ROUTE}
              className="text-blue hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
            >
              About
            </NavLink>
            <NavLink
              to={PROFILE_ROUTE}
              className="hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
              activeClassName="text-darkBlue font-bold"
            >
              Profile
            </NavLink>
          </div>
        </div>
      </nav>
    </>
  )
}
