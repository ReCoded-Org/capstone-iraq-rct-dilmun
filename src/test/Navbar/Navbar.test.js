import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import store from '../../redux/store'
import NavBar from '../../components/Navbar'

it('Test: Navbar Test: ', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={store}>
          <NavBar />
        </Provider>
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
