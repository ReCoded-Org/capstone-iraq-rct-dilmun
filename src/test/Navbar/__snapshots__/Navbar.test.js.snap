// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Navbar Test:  1`] = `
<nav
  className="bg-white z-50 sticky top-0 shadow-md "
>
  <div
    className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
  >
    <div
      className="relative flex items-center justify-between h-16"
    >
      <div
        className="absolute inset-y-0 left-0 flex items-center sm:hidden"
      >
        <button
          aria-controls="mobile-menu"
          aria-expanded="false"
          className="inline-flex items-center justify-center p-2 rounded-md text-blue hover:text-darkBlue  focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Open main menu
          </span>
          <svg
            aria-hidden="true"
            className="block h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
          <svg
            aria-hidden="true"
            className="hidden h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="flex-1 flex items-center ml-8 justify-start sm:items-stretch  "
      >
        <div
          className="flex-shrink-0 flex items-center"
        >
          <img
            alt="logo"
            className="block lg:hidden h-8 w-auto"
            src="logo.svg"
          />
          <img
            alt="logo"
            className="hidden lg:block h-8 w-auto"
            src="logo.svg"
          />
        </div>
        <div
          className="hidden sm:block sm:ml-6"
        >
          <div
            className="flex space-x-4 text-blue"
          >
            <a
              aria-current="page"
              className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium text-darkBlue font-bold"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              navbar.home
            </a>
            <a
              aria-current={null}
              className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
              href="/contact"
              onClick={[Function]}
            >
              navbar.contact
            </a>
            <a
              aria-current={null}
              className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
              href="/about"
              onClick={[Function]}
            >
              navbar.about
            </a>
          </div>
        </div>
      </div>
      <div
        className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
      >
        <button
          className=" hover:bg-darkBlue hover:text-white items-center px-3  space-x-2  text-blue  rounded inline-flex focus:outline-none items-center border border-solid border-blue justify-center"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-globe fa-w-16 "
            data-icon="globe"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <div
            className="text-center "
          >
            langText
          </div>
        </button>
        <div
          className="ml-3 relative"
        >
          <div>
            <button
              className="py-1 px-3 bg-darkBlue  items-center  rounded-full text-white m-1    focus:outline-none"
              onClick={[Function]}
              type="button"
            >
              Log In
            </button>
          </div>
          <div
            aria-labelledby="user-menu"
            aria-orientation="vertical"
            className=" hidden origin-top-right absolute right-0 border border-t-0 boder-grey mt-2 w-48 rounded-md items-center shadow-lg py-1 bg-white ring-1 ring-black ring-opacity-5 focus:outline-none"
            role="menu"
          >
            <a
              aria-current={null}
              className="block px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white"
              href="/profile"
              onClick={[Function]}
              role="menuitem"
            >
              
            </a>
            <button
              className="m-auto text-left block focus:outline-none px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white w-full"
              onClick={[Function]}
              type="button"
            >
              Settings
            </button>
            <button
              className="m-auto text-left block focus:outline-none px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white w-full"
              onClick={[Function]}
              role="menuitem"
              type="button"
            >
              Sign out
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hidden sm:hidden"
    id="mobile-menu"
  >
    <div
      className="px-2 pt-2 pb-3 space-y-1 text-blue"
    >
      <a
        aria-current="page"
        className="text-darkBlue block px-3 py-2 rounded-md text-base font-medium active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        aria-current={null}
        className="text-blue  hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
      <a
        aria-current={null}
        className="text-blue hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        aria-current={null}
        className="hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
        href="/profile"
        onClick={[Function]}
      >
        Profile
      </a>
    </div>
  </div>
</nav>
`;
